.github/workflows/ci.yaml

```diff
@@ -385,31 +385,29 @@ jobs:
         name: hydra-spec
         path: docs/static/
 
-    - name: 📚 Documentation
+    - name: 📚 Documentation sanity check
+      if: github.ref_name != 'master' && github.ref_name != 'release'
       working-directory: docs
       run: |
-        yarn && yarn build
-        mkdir -p public/
-        mv build/* public/
-
-    - name: 💾 Upload docs artifact
-      uses: actions/upload-artifact@v3
-      with:
-        name: docs-stable
```

Comment:
If we never build the docs without unstable, we will never have a "stable starting point" to piece together the docs. I'm afraid these changes would not have the website updated with latest released content and things stick to `0.10.0` or, even worse, try to use a `unstable/` version of the docs (the only we build now) as the stable one?

Reply:
Or am I getting confused by this and it would just work because the `sed` is only done on `master`.. I think this really highlights the point of that this is fragile and we should think about doing it different as described in the red bin item.

Reply:
I'd say you're being confused by the former code building the same doc twice and naming it, once _stable_ and the other time _unstable_. When it's actually more the branch that _decides_ if it's stable or not:
* master: unstable
* release: stable

So, anyway, I decided to go with only one build of the doc instead of two.

What I don't like about this code:
* spaghetti like (we have all the steps and we just activate/deactivate with if and it's hard to follow)
* surprising side effect with the sed you just mentioned

What I like about this code:
* it's not perfect, but it's a bit better that before
* it saves 7 minutes by not building useless stuff
* it works as it used to works

Just saying:  I agree with you @ch1bo , we can probably do better than this P.R. but I would advocate to merge this P.R. because it's making the stuff a little bit better or a little less ugly (depends) than what it was before.